client/state/plans/selectors.js

```diff
@@ -43,7 +48,7 @@ export const getPlan = createSelector(
  */
 export function getPlanRawPrice( state, productId, isMonthly = false ) {
 	const plan = getPlan( state, productId );
-	if ( ! plan || ! ( plan.raw_price || plan.raw_price === 0 ) ) {
+	if ( get( plan, 'raw_price', -1 ) < 0 ) {
```

Comment:
To be honest, this looks rather strange and complicated. I liked your previous solution better. It was more readable. But that's just my opinion and this is a minor detail. :)